sycl/test-e2e/Graph/invalid_profiling.cpp

```diff
@@ -0,0 +1,160 @@
+// REQUIRES: level_zero || cuda, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out 2>&1
+// RUN: %if ext_oneapi_level_zero %{env UR_L0_LEAKS_DEBUG=1 %{run} %t.out 2>&1 | FileCheck --implicit-check-not=LEAK %s %}
+
+// This test checks exception throwing if profiling an event returned
+// from multi-partitions graph submission.
+
+#define GRAPH_E2E_EXPLICIT
+
+#include "graph_common.hpp"
+
+int main() {
+  device Dev;
+  queue Queue{Dev,
+              {sycl::ext::intel::property::queue::no_immediate_command_list{},
+               sycl::property::queue::enable_profiling()}};
+
+  const size_t Size = 100000;
+  int Data[Size] = {0};
+  for (size_t I = 0; I < Size; ++I) {
+    Data[I] = I;
+  }
+  int Values[Size] = {0};
+
+  buffer<int, 1> BufferA(Data, range<1>(Size));
+  buffer<int, 1> BufferB(Values, range<1>(Size));
+  buffer<int, 1> BufferC(Values, range<1>(Size));
+
+  BufferA.set_write_back(false);
+  BufferB.set_write_back(false);
+  BufferC.set_write_back(false);
+  {
+    // kernel launch
+    exp_ext::command_graph KernelGraph{
+        Queue.get_context(),
+        Queue.get_device(),
+        {exp_ext::property::graph::assume_buffer_outlives_graph{}}};
+
+    // Add host_task to create a second partition.
+    auto HostNode = add_node(KernelGraph, Queue,
+                             [&](handler &CGH) { CGH.host_task([=]() {}); });
+
+    auto Nodes = add_kernels(KernelGraph, Size, BufferA, BufferB, BufferC);
+
+    KernelGraph.make_edge(HostNode, Nodes[0]);
+
+    auto KernelGraphExec = KernelGraph.finalize();
+
+    // Run graphs
+    auto KernelEvent = Queue.submit(
```

Comment:
I'd call this variable `GraphEvent`, `KernelEvent` sounds like it's from a single kernel submission rather than an executable graph.